src/main/java/ch/njol/skript/sections/SecConditional.java

```diff
@@ -314,21 +325,27 @@ private Kleenean getHasDelayAfter() {
 		return hasDelayAfter;
 	}
 
+	/**
+	 * Gets the closest conditional section in the list of trigger items
+	 * @param triggerItems the list of items to search for the closest conditional section in
+	 * @param type the type of conditional section to find. if null is provided, any type is allowed.
+	 * @return the closest conditional section
+	 */
 	@Nullable
-	private static SecConditional getClosestIf(List<TriggerItem> triggerItems) {
+	private static SecConditional getPrecedingConditional(List<TriggerItem> triggerItems, @Nullable ConditionalType type) {
 		// loop through the triggerItems in reverse order so that we find the most recent items first
 		for (int i = triggerItems.size() - 1; i >= 0; i--) {
 			TriggerItem triggerItem = triggerItems.get(i);
 			if (triggerItem instanceof SecConditional) {
-				SecConditional secConditional = (SecConditional) triggerItem;
+				SecConditional conditionalSection = (SecConditional) triggerItem;
 
-				if (secConditional.type == ConditionalType.IF)
-					// if the condition is an if, we found our most recent preceding "if"
-					return secConditional;
-				else if (secConditional.type == ConditionalType.ELSE)
+				if (conditionalSection.type == ConditionalType.ELSE)
 					// if the conditional is an else, return null because it belongs to a different condition and ends
 					// this one
 					return null;
+				else if (type == null || conditionalSection.type == type)
```

Comment:
```suggestion
				if (type == null || conditionalSection.type == type)
```
not major but the `else` can be disregarded since the previous `if` returns (and it looks better without the brackets 😉)

Reply:
to the machine it's the same, but imo the `else if` is still beneficial for reading the code. this way, you don't need to read the content of the first `if` to understand that it won't run if the prior conditional is true.

Reply:
fair, id prefer brackets though then since it's a multi line conditional